atlasdb-coordination-impl/src/test/java/com/palantir/atlasdb/coordination/keyvalue/KeyValueServiceCoordinationStoreTest.java

```diff
@@ -159,10 +169,31 @@ public void multipleStoresCanCoexist() {
                         keyValueService,
                         otherCoordinationKey,
                         timestampSequence::incrementAndGet,
-                        String.class);
+                        String.class,
+                        false);
         coordinationStore.transformAgreedValue(unused -> VALUE_1);
         otherCoordinationStore.transformAgreedValue(unused -> VALUE_2);
         assertThat(coordinationStore.getAgreedValue().get().value()).contains(VALUE_1);
         assertThat(otherCoordinationStore.getAgreedValue().get().value()).contains(VALUE_2);
     }
+
+    @Test
+    public void canInitializeAsynchronously() {
+        KeyValueService kvs = mock(KeyValueService.class);
+        doThrow(new IllegalStateException("fail first"))
+                .doAnswer(invocation -> null)
+                .when(kvs)
+                .createTable(any(), any());
+
+        CoordinationStore<String> asyncStore = KeyValueServiceCoordinationStore.create(
+                ObjectMappers.newServerObjectMapper(),
+                kvs,
+                COORDINATION_ROW,
+                timestampSequence::incrementAndGet,
+                String.class,
+                true);
+        Awaitility.waitAtMost(Duration.ONE_MINUTE)
```

Comment:
Is this how we generally test this elsewhere? The alternative would be to just call `tryInitialize` directly, assuming the general async initialization stuff is tested independently.

What's the default wait period for async init? It looks like the default behavior is to wait 10 seconds.

Reply:
Right, the default wait period is 10s. I agree that it's probably better to hit `tryInitialize` for this test, to keep test duration short and avoid waiting 10s - also avoid failures if something else in async init like the standard wait period changes.

Not aware of how this is tested elsewhere apart from integration tests - e.g. `LockStoreImpl`, another use of this, is untested.

Reply:
Post offline discussion - Not sure if this test brings much value, as the framework is tested elsewhere, and this is a straightforward usage (no cleanup tasks), so removing this.

Also taking out Awaitility as we don't use it elsewhere